### DISCUSSION TITLE
Football - Adding Nominating Team as Attribute to Pick Class

### DISCUSSION THREAD

**Discussion by ctmarxen** [cwendt94/espn-api#225] [ESPN API] [answered]
I'd love to determine auction tendencies in terms of how often the nominating team ends up winning the nominated player. Any chance you could add this attribute (i.e. nominating team) to the Pick class so that for any player you can find the team who nominated the player?

**Comment by cwendt94** [comment, marked as answer]
Hey @ctmarxen I'm not 100% familiar with auction drafts (only have experience with auction waiver wire) and currently not part of one. I would love to help and see if this can be added to the Pick class.

Would you be able to get a sample of the draft data? There are two ways you could get the data
1. Can add a print statement [here](https://github.com/cwendt94/espn-api/blob/3fdf52e6eed8a96541e5ef5ee2ffa7a084100ef3/espn_api/football/league.py#L57) of the `picks` object
2. Can run package in debug mode and search the logs for `mDraftDetail`

The first option would be the easiest because in debug mode it will print out all API calls made with their response objects so its a little harder to parse logs.

**Reply by ctmarxen** [reply]
Thanks a ton, @cwendt94! So when I simply run in 'league._fetch_draft()', this is what gets printed (and hence an inspiration for the Pick attribute idea):
<img width="1337" alt="Screen Shot 2021-08-16 at 10 40 44 PM" src="https://user-images.githubusercontent.com/66146717/129655349-ef0d2a5c-a489-46c1-ab81-ece6de91b599.png">

**Reply by ctmarxen** [reply]
Sorry, I'm a rookie dev, but I take it we can use nominatingTeamID to associate that with a team name, and then add that to the Pick class when you're creating it. I haven't tried a pull request, but let me know if you want me to try to do this.

**Reply by cwendt94** [reply]
No worries this is great! So from that example all of the `nominatingTeamID ` equal 0 which does not map to a team. What does that mean for the pick?

To get the team class from the `nominatingTeamID`  you can follow this [line](https://github.com/cwendt94/espn-api/blob/3fdf52e6eed8a96541e5ef5ee2ffa7a084100ef3/espn_api/football/league.py#L59) and add something like this
```python3
nominatingTeam = self.get_team_data(pick['nominatingTeamID'])
```
The value could be `None`

**Reply by ctmarxen** [reply]
Ahh this is great! So the reason why the nominatingTeamID is 0 for those picks is that they were actually keepers (players kept from the previous year). Take a look at the picks once we get beyond the keepers: 
<img width="1342" alt="Screen Shot 2021-08-17 at 11 53 53 AM" src="https://user-images.githubusercontent.com/66146717/129759496-2bd15fbc-e3c8-46e3-80db-09617e5de952.png">

**Reply by ctmarxen** [reply]
I made the updates to league.py and pick.py here: https://github.com/ctmarxen/espn-api/tree/master/espn_api/football 
I don't know how to test the changes though before submitting a pull request. I guess, how do I test the changes locally using this updated api rather than the original one (using Pycharm)?

**Reply by cwendt94** [reply]
That makes sense, I didn't even realize the keeper value had true. You can test locally running this command in the repo
```
python3 setup.py nosetests
```
I will take a look at PR and can give suggestions on how to test

**Reply by ctmarxen** [reply]
Stinking key error, I used 'nominatingTeamID' rather than 'nominatingTeamId'...Hopefully the branch passes the tests now (I also made the changes you suggested)
<img width="579" alt="Screen Shot 2021-08-17 at 11 15 49 PM" src="https://user-images.githubusercontent.com/66146717/129831193-d01bece5-82a5-4126-bc9d-eda07e6255b4.png">
